### DISCUSSION TITLE
Example code of README.md does not work

### DISCUSSION THREAD

**Discussion by iljamaki** [GsActions/commit-message-checker#91] [General]
Hi,

The usage example in the README.me file has two tests:
1. Line length test, max 74 characters per line
2. Linked issue test

I think there are problems with both of them:
1. By experimenting at https://regexr.com/ I found out that the pattern of the first example test only works in the 'Match None' mode, but the action appears to work in the 'Match Any' mode so it is impossible for any valid commit message to pass the test. Is there a way to turn the action into 'Match None' mode to make it work? I did not find such a way by looking into the source code.
2. The regexp pattern in this second example test requires there to be one or more characters in the beginning of the line before the keyword Resolves or Fixes, and also it requires the keyword to be followed by a colon (:) character. Github does not set this kind of requirements, so again the example test will fail valid messages.

I was able to make use of the second example test by modifying its pattern to `^(Resolves|Fixes):? \#[0-9]+$` but I would also like to check the line lengths of commit messages, so any advice to me or bug fix to the action would be highly appreciated.

**Comment by codebydant** [comment]
@iljamaki you can use this pattern: 

```yml
pattern: "^.{0,51}(\n.*)*$"
```

I am testing that the commit message is 50 characters max and is working.

You can also check this search to find regex pattern from other repositories using this github action

https://github.com/search?q=GsActions+%2F+commit-message-checker&type=code

**Reply by iljamaki** [reply]
@dtcMLOps Thank you for the reply and the search link. I think your pattern allows 51 characters on the first line and any number of characters on the following lines (if any). However, with the help of your idea I found the pattern I needed:

`^.{10,50}(\n(\n.{0,72})*)?$`

So this requires the first line to contain 10-50 characters and, if the message continues, the second line must be empty and the following lines 0-72 characters long. The multiline flag is off as newline characters are handled "manually" in the pattern.

README.md should be updated.

**Reply by codebydant** [reply]
Hi @iljamaki I am glad you made it. The README example is just for the subject in the commit message and the body is not included. That's why the `excludeDescription`: 'true' and the `excludeTitle`: 'true' are set.

I have shared a regex pattern for the subject only.

I don't think they will update the README since this repository seems dead.
